### PR TITLE
James 1777 - Retrieve attachment name in GetMessages JMAP method

### PR DESCRIPTION
**Pull request by aduprat**
Rebased of #301, handle attachment name.

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java [hunk 1]
```diff
@@ -125,33 +127,43 @@ public void postDownload(String username, String attachmentId) throws Throwable
...
     }
 
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with \"([^\"]*)\" name$")
+    public void downloadsWithName(String username, String attachmentId, String name) {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
+        AccessToken accessToken = userStepdefs.tokenByUser.get(username);
+        if (accessToken != null) {
+            with().header("Authorization", accessToken.serialize());
+        }
+        response = with().get("/download/" + blobId + "/" + name);
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: It seems this with() is different from the one you add a header.

Comment (aduprat) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: Right

**Comment by rouazana** [comment]
:+1:

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java [hunk 2]
```diff
@@ -72,29 +66,50 @@ public Builder type(String type) {
             return this;
         }
 
+        public Builder name(Optional<String> name) {
+            Preconditions.checkArgument(name != null);
+            this.name = name;
+            return this;
+        }
+
         public Builder size(long size) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java]: you should have removed this method

Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java]: OK

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java [hunk 3]
```diff
@@ -95,7 +87,7 @@ private AttachmentId attachmentId() {
         }
 
         private Long size() {
-            return MoreObjects.firstNonNull(size, Long.valueOf(bytes.length));
+            return Long.valueOf(bytes.length);
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java]: why use a Long ? I guess it can't be null, right ?

Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java]: Right

**Comment by rouazana** [comment]
:+1:

**Comment by mbaechler** [comment]
👍

**Pull request merged**